lib/src/main/java/io/ably/lib/realtime/Channel.java

```diff
@@ -383,8 +375,10 @@ private void onMessage(ProtocolMessage message) {
 			Message msg = messages[i];
 			try {
 				msg.decode(options);
+			} catch (MessageDecodeException e) {
+				Log.e(TAG, String.format("%s on channel %s", e.errorInfo.message, name));
 			} catch(AblyException e) {
-				Log.e(TAG, "Unexpected exception decrypting message", e);
+				Log.e(TAG, String.format("Unexpected exception decrypting message on channel %s", name), e);
```

Comment:
Will this only be a decrypt exception? Ie should it say "Unexpected exception decoding message" ?

Actually, what exception could occur other than a `MessageDecodeException` ?